src/python_testing/TC_TSTAT_4_2.py

```diff
@@ -476,27 +612,13 @@ async def test_TC_TSTAT_4_2(self):
         self.step("17")
         if self.pics_guard(self.check_pics("TSTAT.S.F08") and self.check_pics("TSTAT.S.A0050") and self.check_pics("TSTAT.S.Cfe.Rsp")):
 
-            # Read the PresetTypes to get the preset scenarios supported by the Thermostat.
-            presetTypes = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=cluster.Attributes.PresetTypes)
-
             scenarioNotPresent = None
 
             # Find a preset scenario not present in PresetTypes to run this test.
-            for scenario in cluster.Enums.PresetScenarioEnum:
-                foundMatchingScenario = False
-                for presetType in presetTypes:
-                    if presetType.presetScenario == scenario:
-                        foundMatchingScenario = True
-                        break
-                if not foundMatchingScenario:
-                    scenarioNotPresent = scenario
-                    break
-
-            if scenarioNotPresent is None:
-                logger.info(
-                    "Couldn't run test step 17 since all preset types in PresetScenarioEnum are supported by this Thermostat")
-            else:
-                test_presets = new_presets_with_handle.copy()
+            scenarioMap = dict(map(lambda presetType: (presetType.presetScenario, presetType), presetTypes))
+            unavailableScenarios = list(preset for preset in test_presets if preset.presetScenario not in scenarioMap)
```

Comment:
This doesn't make sense.  `preset.presetScenario` will always be in `scenarioMap`, since presets must have a valid scenario, no?

What's needed is a value of `scenarioNotPresent` (which is not getting set at all by this code!) which is a valid enum value but is not in the `.presetScenario` of anything in `presetTypes`.

Reply:
The current example implementation has every preset scenario, so it's skipping this test. Fixed the test, and removed UserDefined from the server.